anker/src/processor.rs

```diff
@@ -240,6 +310,19 @@ fn process_sell_rewards(program_id: &Pubkey, accounts_raw: &[AccountInfo]) -> Pr
         accounts.anker.key,
         accounts.st_sol_reserve_account,
     )?;
+
+    let clock = Clock::from_account_info(accounts.sysvar_clock)?;
+    let oldest_sample = anker.historical_st_sol_prices.first();
+    let slots_elapsed = clock.slot.saturating_sub(oldest_sample.slot);
+    if slots_elapsed > POOL_PRICE_MAX_SAMPLE_AGE {
+        msg!(
+            "The oldest stSOL/UST price was sampled at slot {}. \
+            It must have been sampled more recently.",
+            oldest_sample.slot,
+        );
+        return Err(AnkerError::FetchPoolPriceNotCalledRecently.into());
+    }
```

Comment:
@enriquefynn I think we also need to enforce a minimum age on the youngest sample. Otherwise you can do the final update and `sell_rewards` in the same transaction. I’ll make a pull request for that.

Reply:
https://github.com/ChorusOne/solido/pull/549

Reply:
Great find! Added comments in the PR